Sources/TuistCore/Xcode/XcodeController.swift

```diff
@@ -34,4 +41,25 @@ public class XcodeController: XcodeControlling {
         }
         return try Xcode.read(path: AbsolutePath(path).parentDirectory.parentDirectory)
     }
+
+    enum XcodeVersionError: Swift.Error {
+        case noXcode
+        case noVersion
+    }
+
+    public func selectedVersion() throws -> Version {
+        guard let xcode = try selected() else {
+            throw XcodeVersionError.noXcode
+        }
+
+        // Xcode versions without patch tag 0 omit it, adding it back if necessary
+        let hasPatchTag = xcode.infoPlist.version.split(separator: ".").count == 3
+        let xcodeVersionString = hasPatchTag ? xcode.infoPlist.version : xcode.infoPlist.version + ".0"
```

Comment:
It has some history, but I agree - to explain `Version` Is copied from SPM; at the time of writing I needed to go around their version that requires 3 components. But yeah, since we now have `Version` In project, it might make sense to change the requirement there.

Reply:
I added extension for `Version` with `init?(unformattedString: String)`. Also added some unit tests for it.